File: Sorting/selection_sort.py
def selectionSort(arr):
    n=len(arr)

    for i in range(n-1):
        min_idx=i
        for j in range(i+1,n):
            if arr[j]<arr[min_idx]:
                min_idx=j
        if(min_idx!=i):
            arr[min_idx],arr[i]=arr[i],arr[min_idx]

if __name__=="__main__":
    arr=[1,2,3,2,2,1,2,3,3,2]

    selectionSort(arr)

    print("Sorted array: ")
    for i in range(len(arr)):
        print("%d"%arr[i], end=" ")

File: Sorting/Selection_sort_.c
#include <stdio.h>

void swap(int *x, int *y)
{
    int temp = *x;
    *x = *y;
    *y = temp;
}

void selectionSort(int arr[], int n)
{
    int i, j, min_idx;
    for (i = 0; i < n - 1; i++)
    {
        min_idx = i;
        for (j = i + 1; j < n; j++)
        {

            if (arr[j] < arr[min_idx])
                min_idx = j;
        }

        if (min_idx != i)
            swap(&arr[min_idx], &arr[i]);
    }
}

void printArray(int arr[], int size)
{
    for (int i = 0; i < size; i++)
    {
        printf("%d ", arr[i]);
    }
    printf("\n");
}
int main()
{
    int arr[] = {1, 1, 2, 3, 2, 3, 1, 1, 3, 2};
    int n = sizeof(arr) / sizeof(arr[0]);

    selectionSort(arr, n);
    printf("Sorted Array:\n");
    printArray(arr, n);
    return 0;
}